src/tracker/mime-type-tracker-query.vala

```diff
@@ -34,13 +38,40 @@ public class Games.MimeTypeTrackerQuery : Object, TrackerQuery {
 		return false;
 	}
 
-	public Game game_for_cursor (Tracker.Sparql.Cursor cursor) throws Error {
+	public void process_cursor (Tracker.Sparql.Cursor cursor) {
 		var uri = cursor.get_string (0);
+		uris += uri;
+	}
+
+	public async void foreach_game (GameCallback game_callback) {
+		uint handled_uris = 0;
+		foreach (var uri in uris) {
+			try {
+				var game = game_for_uri (uri);
```

Comment:
It's considered a good practice to only enclose code throwing errors in a try blocks (or barely more).